### PR TITLE
Cleanup + a couple of little bugs

### PR DESCRIPTION
**Pull request by gytis-ivaskevicius**

### CODE REVIEW

File: examples/fully-featured/myNixpkgsPatch.patch [hunk 1]
```diff
@@ -0,0 +1,12 @@
+diff --git a/pkgs/top-level/all-packages.nix b/pkgs/top-level/all-packages.nix
+index 6dd393492c8..c8456afc28d 100644
+--- a/pkgs/top-level/all-packages.nix
++++ b/pkgs/top-level/all-packages.nix
+@@ -30930,4 +30930,6 @@ in
+   lc3tools = callPackage ../development/tools/lc3tools {};
+
+   zktree = callPackage ../applications/misc/zktree {};
++
++  flake-utils-plus-test = callPackage ../applications/misc/hello { };
```
Comment (gytis-ivaskevicius) [inline examples/fully-featured/myNixpkgsPatch.patch:10]: Actually a proper patch to test patching functionality

File: repl.nix [hunk 2]
```diff
@@ -6,6 +6,6 @@ in
 { inherit flake; }
 // flake
 // builtins
-// nixpkgs
+// flake.nixosConfigurations.${hostname}
 // nixpkgs.lib
-  // flake.nixosConfigurations.${hostname}
+  // nixpkgs
```
Comment (gytis-ivaskevicius) [inline repl.nix:11]: There was an issue that `flake.nixosConfigurations.<hostname>.pkgs` were overwriting `nixpkgs` which was not intended

File: systemFlake.nix [hunk 3]
```diff
@@ -9,7 +9,6 @@
 , channelsConfig ? { }
 , sharedOverlays ? [ ]
 
-, nixosConfigurations ? { } # deprecate soon, no longer used or works
```
Comment (gytis-ivaskevicius) [inline systemFlake.nix:12]: It was unused

File: systemFlake.nix [hunk 4]
```diff
@@ -179,6 +168,6 @@ otherArguments
   // optional checksBuilder { checks = checksBuilder pkgs; }
```
Comment (gytis-ivaskevicius) [inline systemFlake.nix:168]: There is an issue of builders not getting merged with actual arguments :/ I'd love to see such use-case working:
```patch
diff --git a/examples/fully-featured/flake.nix b/examples/fully-featured/flake.nix
index 5eb7ade..13071f9 100644
--- a/examples/fully-featured/flake.nix
+++ b/examples/fully-featured/flake.nix
@@ -139,6 +139,7 @@

       # All other values gets passed down to the flake
       overlay = import ./overlays;
+      packages.x86_64-linux.merge-with-packagesBuilder-test = self.pkgs.x86_64-linux.nixpkgs.hello;
       abc = 132;
       # etc
 ```
 Maybe you guys could think of some clean merge function for this? My solution would be quite lenghty

Comment (gytis-ivaskevicius) [reply systemFlake.nix:168]: Dumb and simple idea came to my mind:
 `mapAttrs (system: _: packages1.${system} or {} // packages2.${system} or {} ) (packages1 // packages2)` 

looks good?

Comment (fufexan) [reply systemFlake.nix:168]: There may be the same issue with `appsBuilder` and `apps`. Should probably fix that as well.

Comment (gytis-ivaskevicius) [reply systemFlake.nix:168]: Yes, there are. Everything that depends on `<system>` is implemented in the same manner, but also fix is the same for all of these properties as well

**Pull request merged**